blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/BranchContainerImpl.java

```diff
@@ -145,15 +145,20 @@ public BluePipeline get(String name) {
     @Override
     @SuppressWarnings("unchecked")
     public Iterator<BluePipeline> iterator(int start, int limit) {
-        List<BluePipeline> branches = new ArrayList<>();
-        Collection<Job> jobs = pipeline.mbp.getAllJobs();
-
-        jobs = ContainerFilter.filter(jobs, start, limit);
-        for(Job j: jobs){
-            branches.add(new BranchImpl(j, getLink()));
-        }
-
-        return Ordering.from(BRANCH_COMPARITOR).sortedCopy(branches).iterator();
+        final Link link = getLink();
+        // Filter will decide if the requester wants branches or pull requests
+        Collection allJobsMatchinFilter = ContainerFilter.filter(pipeline.mbp.getAllJobs());
```

Comment:
Looks like, we are doing things differently than how classic does.  See `BranchCategoryFilter.filter()` and `ChangeRequestSCMHeadCategory.isMatch()`. As far as I can see they are not using action to determine pull request. 

Here is the stapler stack to render Pull Request tab:

```
Stapler-Trace-001:-> evaluate(<hudson.model.Hudson@6ee3aead> :hudson.model.Hudson,"/job/pipeline-demo-test/view/change-requests")
Stapler-Trace-002:-> evaluate(((StaplerProxy)<hudson.model.Hudson@6ee3aead>).getTarget(),"/job/pipeline-demo-test/view/change-requests")
Stapler-Trace-003:-> evaluate(<hudson.model.Hudson@6ee3aead>.getJob("pipeline-demo-test"),"/view/change-requests")
Stapler-Trace-004:-> evaluate(<org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject@34b9e089[pipeline-demo-test]> :org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject,"/view/change-requests")
Stapler-Trace-005:-> evaluate(<org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject@34b9e089[pipeline-demo-test]>.getView("change-requests"),"")
Stapler-Trace-006:-> evaluate(<jenkins.branch.MultiBranchProjectViewHolder$ViewImpl@264ba7e5[job/pipeline-demo-test/view/change-requests/]> :jenkins.branch.MultiBranchProjectViewHolder$ViewImpl,"")
Stapler-Trace-007:-> hudson/model/View/index.jelly on <jenkins.branch.MultiBranchProjectViewHolder$ViewImpl@264ba7e5[job/pipeline-demo-test/view/change-requests/]>
```